lib/src/layer/marker_layer.dart

```diff
@@ -1,75 +1,107 @@
-        top = _topOffset(height, alignOpt);
-
-  static double _leftOffset(double width, AnchorAlign alignOpt) =>
-      switch (alignOpt) {
-        AnchorAlign.left => 0,
-        AnchorAlign.right => width,
-        _ => width / 2,
-      };
-
-  static double _topOffset(double height, AnchorAlign alignOpt) =>
-      switch (alignOpt) {
-        AnchorAlign.top => 0,
-        AnchorAlign.bottom => height,
-        _ => height / 2,
-      };
-
-  factory Anchor.forPos(AnchorPos? pos, double width, double height) {
-    if (pos == null) return Anchor._(width, height, AnchorAlign.none);
-    if (pos.alignment case final align?) return Anchor._(width, height, align);
+  factory Anchor.fromPos(AnchorPos pos, double width, double height) {
     if (pos.anchor case final anchor?) return anchor;
+    if (pos.alignment case final alignment?) {
+      return Anchor(
+        switch (alignment._x) { -1 => 0, 1 => width, _ => width / 2 },
+        switch (alignment._y) { 1 => 0, -1 => height, _ => height / 2 },
+      );
+    }
     throw Exception();
   }
 }
 
-class AnchorPos {
-  final Anchor? anchor;
-  final AnchorAlign? alignment;
+/// Relative alignment for a [Marker.builder] widget relative to the center
+/// of its bounding box defined by its [Marker.height] & [Marker.width]
+enum AnchorAlign {
+  topLeft(-1, 1),
+  topRight(1, 1),
+  bottomLeft(-1, -1),
+  bottomRight(1, -1),
 
-  AnchorPos.exactly(this.anchor) : alignment = null;
-  AnchorPos.align(this.alignment) : anchor = null;
-}
+  center(0, 0),
 
-enum AnchorAlign {
-  none,
-  left,
-  right,
-  top,
-  bottom,
-  center,
+  /// Top center
+  top(0, 1),
+
+  /// Bottom center
+  bottom(0, -1),
+
+  /// Left center
+  left(-1, 0),
+
+  /// Right center
+  right(1, 0),
+
+  @Deprecated(
+    'Prefer `center`. '
+    'This value is equivalent to the `center` alignment. '
+    'If you notice a difference in behaviour, please open a bug report on GitHub. '
+    'This feature is deprecated since v5.',
+  )
+  none(0, 0);
+
+  final int _x;
+  final int _y;
+
+  const AnchorAlign(this._x, this._y);
 }
 
-/// Marker object that is rendered by [MarkerLayerWidget]
+/// Represents a coordinate point on the map with an attached widget [builder],
+/// rendered by [MarkerLayer]
+///
+/// Some properties defaults will absorb the values from the parent [MarkerLayer],
+/// if the reflected properties are defined there.
 class Marker {
+  final Key? key;
+
   /// Coordinates of the marker
   final LatLng point;
 
   /// Function that builds UI of the marker
   final WidgetBuilder builder;
-  final Key? key;
 
   /// Bounding box width of the marker
   final double width;
 
   /// Bounding box height of the marker
   final double height;
-  final Anchor anchor;
 
-  /// If true marker will be counter rotated to the map rotation
+  /// Positioning of the [builder] widget relative to the center of its bounding
+  /// box defined by its [height] & [width]
+  final AnchorPos? anchorPos;
```

Comment:
Hi,

I changed it because the constructor used `AnchorPos` anyway, and because I added the ability for `MarkerLayer` to override the anchor position of its descendants. There was also a few other changes that meant this was the best way of doing things.
I didn't think about the consequences of this causing a new calculation every rebuild though. Does this cause a significant performance disadvantage?

It's now nullable because `MarkerLayer` is now able to override it. If you leave it null, it will default to anchoring at the center.

(Apologies for the screenshot instead of line refs, I'm on mobile)

![Screenshot_2023-06-07-07-45-43-868_com.github.android.png](https://github.com/fleaflet/flutter_map/assets/58115698/4eadd876-89a9-4251-a507-fc4cbfe97633)

Reply:
>Does this cause a significant performance disadvantage?

Honestly I don't feel like I have a reliable way to measure this, so far my attempts at performance comparisons with flutter_map feel quite random.

>Can you post your code snippet that you're having trouble migrating?

https://github.com/rorystephenson/flutter_map_marker_popup/blob/8139b1874141885a04bd95d3e2b089a64eea1fd7/lib/src/marker_layer.dart#L80-L83

Here you can see where I used the calculated anchor value, I am able to migrate but I must now calculate the anchor on each build and this requires assuming a default anchorPos if none is defined.

---

We can support defining the anchorPos on MarkerLayer and still store the anchor on Markers if the user defines it by just storing the `Anchor?` on Marker instead of `AnchorPos` and only setting it when the user provides an `anchorPos` on the Marker. However I feel this hides the performance implications (it is more efficient to define anchor per-Marker than at the layer level) from the users. I definitely see the appeal of being able to define such options at the MarkerLayer level (like rotation etc) however for parameters which we can pre-calculate, like Anchor, I'm not sure it's a worthwhile tradeoff.

Reply:
I've opened an issue to discuss this further: https://github.com/fleaflet/flutter_map/issues/1552